packages/server/src/api/controllers/static/index.ts

```diff
@@ -397,3 +518,75 @@ export const getSignedUploadURL = async function (
 
   ctx.body = { signedUrl, publicUrl }
 }
+
+export async function servePwaManifest(ctx: UserCtx<void, any>) {
+  const appId = context.getAppId()
+  if (!appId) {
+    ctx.status = 404
+    ctx.body = { message: "App not found" }
+    return
+  }
+
+  try {
+    const db = context.getAppDB({ skip_setup: true })
+    const appInfo = await db.get<App>(DocumentType.APP_METADATA)
+
+    if (!appInfo.pwa) {
+      ctx.status = 404
+      ctx.body = { message: "PWA not configured for this app" }
+      return
+    }
+
+    const manifest: PWAManifest = {
+      name: appInfo.pwa.name || appInfo.name,
+      short_name: appInfo.pwa.short_name || appInfo.name,
+      description: appInfo.pwa.description || "",
+      start_url: `/app${appInfo.url}`,
+      display: appInfo.pwa.display || "standalone",
+      background_color: appInfo.pwa.background_color || "#FFFFFF",
+      theme_color: appInfo.pwa.theme_color || "#FFFFFF",
+      icons: [],
+      screenshots: [],
+    }
+
+    if (appInfo.pwa.icons && appInfo.pwa.icons.length > 0) {
+      try {
+        manifest.icons = await objectStore.enrichPWAImages(appInfo.pwa.icons)
+
+        const desktopScreenshot = manifest.icons.find(
+          icon => icon.sizes === "1240x600" || icon.sizes === "2480x1200"
+        )
+        if (desktopScreenshot) {
+          manifest.screenshots?.push({
```

Comment:
This will not push the object if `screenshots` is undefined. Is this what you are attempting?
Otherwise, you would need to ensure that `manifest.screenshots` exists: `manifest.screenshots ??= []`

Reply:
Screenshots is always initialised as an empty array in the manifest definition on line 559.

Reply:
I see, so probably the `?` is not required